packages/aws-cdk/lib/api/hotswap-deployments.ts

```diff
@@ -43,9 +44,15 @@ export async function tryHotswapDeployment(
     listStackResources,
   });
 
-  const currentTemplate = await cloudFormationStack.template();
-  const stackChanges = cfn_diff.diffTemplate(currentTemplate, stackArtifact.template);
-  const hotswappableChanges = await findAllHotswappableChanges(stackChanges, evaluateCfnTemplate);
+  const cloudFormationDeployments = new CloudFormationDeployments({
+    sdkProvider,
+  });
```

Comment:
Do you mean to extract `readCurrentTemplateWithNestedStacks()` and all the private helpers we created for it to a new class (`NestedStackOperationsHelper`?) and then changing `CloudFormationDeployments.readCurrentTemplateWithNestedStacks()` to be a one-liner that returns void and calls `NestedStackOperationsHelper.readCurrentTemplateWithNestedStacks()`  -- and calling `NestedStackOperationsHelper.readCurrentTemplateWithNestedStacks()` from `tryHotswapDeployment()`?

Reply:
Probably not all private helpers (for example, `prepareSdkWithLookupOrDeployRole()` should most likely stay in `CloudFormationDeployments`), and I'm not sure it's obvious it should be a class, vs. a function, for example. I leave those details up to you.

But yes, that's what I meant 🙂.

Reply:
Also, I assume this is just a simple typo from writing these comments fast, but no, `CloudFormationDeployments.readCurrentTemplateWithNestedStacks()` should not return `void` after this refactoring 🙂. It should return exactly what it does now. It's just that its implementation will be moved.

Reply:
Our new helper needs to call some methods from `CloudFormationDeployments`, namely `readCurrentTemplate()`. Should we pass an instance of `CloudFormationDeployments` to our helper so it can call `readCurrentTemplate()`, or is there a better way? Passing that instance means a `CloudFormationDeployments` has to be passed through the entire call chain you listed above.

Reply:
No. We would need to move the implementation of those methods like `readCurrentTemplate()` also to this new helper, exactly like we do for `readCurrentTemplateWithNestedStacks()` (leave the API in `CloudFormationDeployments` the same, just make `CloudFormationDeployments.readCurrentTemplate()` call this new helper in its implementation).